packages/material-ui/src/Select/SelectInput.js

```diff
@@ -360,9 +360,11 @@ SelectInput.propTypes = {
   /**
    * The icon that displays the arrow.
    */
-  IconComponent: PropTypes.elementType,
+  IconComponent: PropTypes.elementType.isRequired,
   /**
    * Use that prop to pass a ref to the native select element.
+   *
+   * TODO: there's no native select element here
```

Comment:
I need to investigate how this is used. As far as I know this is a private component. There's no "native select element" in this component and even worse: `inputRef.current.focus()` focuses the trigger and not the `listbox` (which is closer to the native `select` IMO).

Reply:
So `inputRef` is not documented for `Select` but `Input`. The documentation for `SelectInput` is ignored anyway. 

`inputRef` was added in https://github.com/mui-org/material-ui/pull/10070/files to support integration with `downshift` when we still rendered a native `select`. Currently `SelectInput` isn't even used in those demos anymore. So over time the original intention got lost.

I think for v5 we need to evaluate what this prop tries to achieve. If it is focus we can rename it. If it is the value we can use a custom interface for the `{ value: string }` interface.

Reply:
Right now these are equivalent:

```jsx
<Select inputProps={{ ref }} />;
<Select inputProps={{ inputRef: ref }} />;
```
where `ref` will be passed an implementation of `{ focus(): void; node: HTMLElement; value: T }`. I don't think this is something anybody would want to use. This is why we need to ask for use cases and add tests and not just blindly accept "I need X prop on Y component, thanks".

Reply:
If I remember correctly, `NativeSelect` was introduced after the `native` prop, to solve a tree shaking issue.

The TextField abstracts the different form components into a single one. I think that we need this sort of abstraction for the text inputs and for the select. Should it be carried by the TextField component? I don't know. It's important to reduce boilerplate, in 80% of the time, people don't need to pull out and tweak each component independently.

Reply:
My question would be why you think a TextField needs handle the select UI but not checkbox, radio etc. This is an incredibly bloated component that does way more than it should do at the cost of having a huge bundle size just so that you can type `<TextField select />` instead of `<Select />`.

If you write UIs and think a  text field and select are somewhat interchangeable then you have way more problems beyond performance.

It all comes down to thinking they should have the same implementation/behavior because the material guidelines make them look similar. Speaking of which: I couldn't find them in the guidelines. I can identify them in some of the Text Fields illustrations but there's no mention and their UX does not match the description of the Text Fields which speak of "editing text".

Reply:
I think that the we need an abstraction to glue a FormControl, FormLabel, Select, FormHelperText together. Is the TextField the best host for it? That, I don't know.